python/pyspark/sql/dataframe.py

```diff
@@ -861,11 +861,11 @@ def groupBy(self, *cols):
             Each element should be a column name (string) or an expression (:class:`Column`).
 
         >>> df.groupBy().avg().collect()
-        [Row(avg(age)=3.5)]
+        [Row(avg_c1=3.5)]
```

Comment:
@cloud-fan Actually, initially i had it scoped only to wrap the expression with UnresolvedAlias for TypedAggregateExpression. I then stumbled up on this  [test](https://github.com/apache/spark/blob/master/sql/hive/src/test/scala/org/apache/spark/sql/hive/parquetSuites.scala#L672-#L687). 

Wenchen, what do you think about the above test ? Basically this prompted me to widen the scope to generate simpler and shorter (i agree, its not better) for AggregateExpressions instead of limiting it to typed aggregates.